File: src/MainClasses/Comment.java
package MainClasses;

import Exceptions.FieldIsEmptyException;

public class Comment extends BaseClass<Comment> {

    private String ownerId;

    private String script;

    private String postId;


    public Comment(String ownerId, String script, String photoId) {
        this.ownerId = ownerId;
        this.script = script;
        this.postId = photoId;
    }

    public String getOwnerId() {
        return ownerId;
    }

    public void setOwnerId(String ownerId) {
        this.ownerId = ownerId;
    }

    public String getScript() {
        return script;
    }

    public void setScript(String script) {
        this.script = script;
    }

    public String getPostId() {
        return postId;
    }

    public void setPostId(String postId) {
        this.postId = postId;
    }


    public void validate() {
        if(script == null){
            throw new NullPointerException("script is required.");
        }
        if(script.isEmpty()){
            throw new FieldIsEmptyException("script cannot be empty!","script");
        }
    }
    public void editComment(String script){
        if(script.isEmpty()){
            throw new FieldIsEmptyException("script cannot be empty!","script");
        }
        this.script = script;
    }
}

File: frontend/lib/models/comment.dart
class Comment {
  final String id;
  final String ownerId;
  final String script;
  final String postId;

  const Comment({
    required this.id,
    required this.ownerId,
    required this.script,
    required this.postId,
  });

  Comment copyWith({
    String? id,
    String? ownerId,
    String? script,
    String? postId,
  }) {
    return Comment(
      id: id ?? this.id,
      ownerId: ownerId ?? this.ownerId,
      script: script ?? this.script,
      postId: postId ?? this.postId,
    );
  }

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: (json['id'] ?? '').toString(),
      ownerId: (json['ownerId'] ?? '').toString(),
      script: (json['script'] ?? '').toString(),
      postId: (json['postId'] ?? '').toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'ownerId': ownerId,
      'script': script,
      'postId': postId,
    };
  }
}
